File: задача2_решение.md/SubscriberViewer.py
from enum import Enum

class SocialNetwork(Enum):
    Instagram = 1
    Twitter = 2

class SubscriberViewer:
    def __init__(self):
        self._strategies = {
            SocialNetwork.Instagram: InstagramStrategy(),
            SocialNetwork.Twitter: TwitterStrategy()
        }
    
    def get_subscribers(self, user_name: str, network_type: SocialNetwork) -> List['SocialNetworkUser']:
        strategy = self._strategies.get(network_type)
        if not strategy:
            raise ValueError("Unsupported social network")
        return strategy.get_subscribers(user_name)

File: задача2_решение.md/SubscriberViewer.cs
public class SubscriberViewer
{
    private readonly Dictionary<SocialNetwork, ISocialNetworkStrategy> _strategies;

    public SubscriberViewer()
    {
        _strategies = new Dictionary<SocialNetwork, ISocialNetworkStrategy>
        {
            { SocialNetwork.Instagram, new InstagramStrategy() },
            { SocialNetwork.Twitter, new TwitterStrategy() }
        };
    }

    public SocialNetworkUser[] GetSubscribers(string userName, SocialNetwork networkType)
    {
        if (!_strategies.TryGetValue(networkType, out var strategy))
            throw new ArgumentException("Unsupported social network");

        return strategy.GetSubscribers(userName);
    }
}